src/what-season.js

```diff
@@ -1,6 +1,41 @@
 const CustomError = require("../extensions/custom-error");
 
-module.exports = function getSeason(/* date */) {
-  throw new CustomError('Not implemented');
-  // remove line with error and write your code here
+module.exports = function getSeason(userDate) {
+  if (typeof(userDate) === 'undefined' || userDate === null)
+  {
+    return 'Unable to determine the time of year!';
+  }
+  if (this.toString.call(userDate) === '[object Date]')
+  {
+    if (isNaN(userDate.getTime()))
+    {
+      throw CustomError(userDate + 'is not date format');
+    }
+  }
+  else
+    {
+      throw CustomError('Unable to determine the time of year!');
+  }
+
+  let month = userDate.getMonth();
+  let winter = 'winter';
+  let spring = 'spring';
+  let summer = 'summer';
+  let fall = 'fall';
+  let seasons = {
```

Comment:
I would make function instead this to not duplicate values. You can use switcher inside:
const getSeason = index => {
swith(index) {
case 12,0,1,2: return winter;
case 3,4,5,6: return sprint;
...etc.